pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -131,10 +143,24 @@ func (p *criStatsProvider) ListPodStats() ([]statsapi.PodStats, error) {
 		// container belongs to.
 		ps, found := sandboxIDToPodStats[podSandboxID]
 		if !found {
+			caPodSandbox, found := caInfos[podSandboxID]
+			if !found {
+				glog.Errorf("Unable to find cadvisor stats for sandbox %q", podSandboxID)
+			}
 			ps = p.makePodStats(podSandbox)
+			// Fill stats from cadvisor is available for full set of required pod stats
+			if found {
+				p.addCadvisorPodStats(ps, &caPodSandbox)
+			}
 			sandboxIDToPodStats[podSandboxID] = ps
 		}
-		ps.Containers = append(ps.Containers, *p.makeContainerStats(stats, container, &rootFsInfo, uuidToFsInfo))
+		cs := p.makeContainerStats(stats, container, &rootFsInfo, uuidToFsInfo)
+		// If cadvisor stats is available for the container, use it to populate
+		// container stats
+		if caFound {
+			p.addCadvisorContainerStats(cs, &caStats)
+		}
```

Comment:
Should we also report error here? Or should we remove the error above?

Reply:
There is no error returned from this right ? This only parses the stats and adds the available one's. The only non disrupting error will be if we dont find cadvisor stats which needs to be printed above?

Reply:
I see.